### ISSUE TITLE
Passing a classname to an areachart dot does not assign it.

### ISSUE THREAD

**Issue by 108yen** [recharts/recharts#4290]
- [x] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://github.com/108yen/recharts](https://github.com/108yen/recharts)

### Steps to reproduce
Passing className to areachart dot
```tsx
          <Area
            type="monotone"
            dot={{ className： 'custom-dot-className' }}
            dataKey="uv"
            stroke="#8884d8"
            fill="#8884d8"
          />
```

### What is expected?
The className is assigned to the areaChart dot.

### What is actually happening?
The className is not assigned to the areaChart dot.

| Environment | Info |
|---|---|
| Recharts | v2.12.2 |
| React | 16.14.0 |
| System | windows |
| Browser | chrome |

**Comment by 108yen** [comment]
Can I fix this issue?
Refer to Line.tsx.

**Comment by ckifer** [comment]
Merged in you PR, much appreciated! This is in the 3.x branch but can be easily backported to master/2.x as well. @108yen if this is blocking you, could you merge the commit/make the same PR to the master branch as well?

**Comment by 108yen** [comment]
@ckifer 
Thank you!
I will fix the master branch as well!
I'm going to sleep now, so it will be tomorrow!

**Comment by ckifer** [comment]
Going to close this for now, will be released in the next patch

**Comment by ckifer** [comment]
released in 2.12.3
